### PR TITLE
chore: speed up ci flow

### PR DESCRIPTION
**Pull request by aleksuss**
The goal of the PR is to speed up CI execution by delegating running app test to the self-hosted runners.

### CODE REVIEW

**Comment by birchmd** [comment]
I think a lot of the CI time is spent on compiling nearcore from scratch. Probably the best way to improve the speed would be to cache the nearcore binary and only recompile when we update what version we depend on.

File: etc/integration-tests/src/nearcore_utils.rs [hunk 1]
```diff
@@ -113,3 +119,50 @@ pub async fn start_neard(
 
     Ok(child)
 }
+
+pub fn neard_path() -> anyhow::Result<PathBuf> {
+    std::env::current_dir()
```
Comment (birchmd) [inline etc/integration-tests/src/nearcore_utils.rs:124]: I wonder if instead of the current directory we should use `refiner_utils::get_repository_root()`

**Review by birchmd — DISMISSED** [review]

File: Makefile [hunk 2]
```diff
@@ -12,7 +12,10 @@ check-compile:
 	$(CARGO) check --all-targets
 
 test:
-	$(CARGO) test
+	$(CARGO) test --workspace --exclude aurora-refiner-app-integration-tests
+
+run-test:
```
Comment (birchmd) [inline Makefile:17]: Maybe calling this `integration-test` would be more descriptive

File: .github/workflows/rust.yml [hunk 3]
```diff
@@ -36,6 +36,16 @@ jobs:
     steps:
       - name: Checkout sources
         uses: actions/checkout@v2
+      - name: Cargo Cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/bin/
```
Comment (birchmd) [inline .github/workflows/rust.yml:43]: Does caching a home directory create any race conditions when multiple runners are going at the same time?

**Review by birchmd — DISMISSED** [review]

File: .github/workflows/rust.yml (hunk 3 above)
Comment (aleksuss) [reply .github/workflows/rust.yml:43]: I suppose there is some kind of lock that exists in another way it doesn't make sense.

**Review by birchmd — APPROVED** [review]

**Pull request merged**